ehrql/utils/sequence_utils.py

```diff
@@ -0,0 +1,18 @@
+def ordered_set(sequence):
+    """
+    Deduplicates a sequence, maintaining order
+    """
+    return list(dict.fromkeys(sequence))
+
+
+def get_grouping_level_as_int(all_groups, group_subset):
```

Comment:
This isn't a sequence util really, but I've put it here until I work out how to categorise it

Reply:
Finding appropriate locations for the random utility functions you end up needing is one of the genuinely hard problems in computer science.

You could arguably stick it in `math_utils` I mean, it's got numbers in ,right?